packages/generator/ni_measurement_plugin_sdk_generator/client/_support.py

```diff
@@ -284,3 +323,26 @@ def _get_python_type_as_str(type: Field.Kind.ValueType, is_array: bool) -> str:
     if is_array:
         return f"List[{python_type.__name__}]"
     return python_type.__name__
+
+
+def _get_enum_type(parameter: Any, enum_values_by_type_name: Dict[str, Dict[str, Any]]) -> Any:
+    if parameter.type == FieldDescriptorProto.TYPE_ENUM:
+        loaded_enum_values = json.loads(parameter.annotations["ni/enum.values"])
+        enum_values = dict((key, value) for key, value in loaded_enum_values.items())
+
+        for existing_enum_name, existing_enum_values in enum_values_by_type_name.items():
+            if existing_enum_values == enum_values:
+                return Enum(existing_enum_name, existing_enum_values)
```

Comment:
It's kind of weird to create the same enum type more than once. I'm not sure whether this returns the same class object or a new one. It would be less weird to cache the enum types in the dict too.

Reply:
Thanks for pointing this out. I've refined this method such that we cache the created enum also, so that we won't have to create multiple enums.
Updated:
```
enum_values_by_type: Dict[str], Dict[str, int]]  -> enum_values_by_type: Dict[Type[Enum], Dict[str, int]] 
```